server/src/test/java/org/apache/druid/server/http/DataSourcesResourceTest.java

```diff
@@ -182,9 +182,9 @@ public void testGetFullQueryableDataSources()
     Set<ImmutableDruidDataSource> result = (Set<ImmutableDruidDataSource>) response.getEntity();
     Assert.assertEquals(200, response.getStatus());
     Assert.assertEquals(2, result.size());
-    Assert.assertEquals(
-        listDataSources.stream().map(DruidDataSource::toImmutableDruidDataSource).collect(Collectors.toSet()),
-        new HashSet<>(result)
+    TestUtils.assertEqualsImmutableDruidDataSource(
+        listDataSources.stream().map(DruidDataSource::toImmutableDruidDataSource).collect(Collectors.toList()),
```

Comment:
This test code depends on how `listDataSources` is created and it works only when `listDataSources` is in the same order with `result`. They should be sorted first before you compare them.

Reply:
@jihoonson , Got it. But the original test code uses Collectors.toSet() which is an unordered collector. How was the sorting order ensured in the actual code ? Correct me if my understanding is wrong.

Reply:
@jihoonson , @leventov Please check the above.

Reply:
The "equality" is checked differently for `ArrayList` and `HashSet`. Set is not an ordered collection, so ordering is not considered when comparing two sets while List is an ordered one. Please check the code of `ArrayList.equals()` and `HashSet.equals()` for details.

Reply:
Fixed. ImmutableDruidDataSourceTestUtils.assertEqualsImmutableDruidDataSource(List<>, List<> ) takes care of this.  Direct sort() methods on ImmutableDruidDataSource collection cannot be used as equals() and hashCode() methods are made as unsupported operations. Instead the two lists are compared in O(n^2). This should be fine I think for these test cases as the size of the lists is not greater than 2.